### ISSUE TITLE
async-std@1.6-beta.1: wasm-timer `Delay` pauses

### ISSUE THREAD

**Issue by azriel91** [async-rs/async-std#775]
Heya, e4df140 uses `wasm_timer::Delay` in `src/utils.rs` for WASM target's `Timer` implementation. When running a countdown async task in a loop with a 50 ms delay, it pauses for 4 seconds every 19 iterations (observed in both Chrome and Firefox):

![delay_wasm_timer](https://user-images.githubusercontent.com/2993230/81750527-c7558d00-9501-11ea-9c5a-0c64dc4ed733.gif)

Switching it to `futures_timer::Delay` allows it to loop smoothly:

![delay_futures_timer](https://user-images.githubusercontent.com/2993230/81750569-e05e3e00-9501-11ea-964c-bf6d13357414.gif)

#776 has the change, though `futures-timer` depends on `async-std` in `[dev-dependencies]`, so not sure if you'd like the somewhat cyclic reference.

[`futures-timer`](https://crates.io/crates/futures-timer) backs onto [`gloo_timers`](https://crates.io/crates/gloo-timers), which [calls the browser's `setTimeout`](https://github.com/rustwasm/gloo/blob/master/crates/timers/src/callback.rs#L58) API. I couldn't figure out how `wasm-timer` does it.

**Comment by dignifiedquire** [comment]
Interesting, futures-timer was a dependency before in here, so this should be fine

**Comment by Fishrock123** [comment]
Perhaps there is some misunderstanding, but `wasm-timer` still [uses javascript `setTimeout`](https://github.com/tomaka/wasm-timer/blob/master/src/timer/global/wasm.rs#L29).

The bug is probably due to this unwrap failing and defaulting to 5 seconds: https://github.com/tomaka/wasm-timer/blob/master/src/timer/global/wasm.rs#L60

**Comment by azriel91** [comment]
> Perhaps there is some misunderstanding, but `wasm-timer` still [uses javascript `setTimeout`](https://github.com/tomaka/wasm-timer/blob/master/src/timer/global/wasm.rs#L29).

ah, I didn't `grep` the `wasm-timer` repository; did human-eye file traversal, and gave up because I went too far down the (wrong) rabbit hole.

I like `futures-timer`'s simplicity. If deciding whether or not to use `wasm-timer`, one would have to:

* understand the value the additional complexity provides
* figure out why that `unwrap` is failing

**Comment by tomaka** [comment]
Author of `wasm-timer` here:
To create `wasm-timer` I took what was `futures-timer` at the time and changed "the backend".

The reason why I didn't do it the way `futures-timer 0.3` does is because the public-facing API at the time was exposing the `Instant` and `Interval` types. I had to expose a custom version of `Instant`, and `Interval` is quite complicated to implement.

Nowadays there's no good reason to use `wasm_timer` anymore.
